zebra-rpc/src/server/tracing_middleware.rs

```diff
@@ -0,0 +1,74 @@
+//! A custom middleware to trace unrecognized RPC requests.
+
+use std::future::Future;
+
+use futures::future::{Either, FutureExt};
+use jsonrpc_core::{
+    middleware::Middleware,
+    types::{Call, Failure, Output, Response},
+    BoxFuture, Error, ErrorCode, Metadata, MethodCall, Notification,
+};
+
+/// A custom RPC middleware that logs unrecognized RPC requests.
+pub struct TracingMiddleware;
+
+impl<M: Metadata> Middleware<M> for TracingMiddleware {
+    type Future = BoxFuture<Option<Response>>;
+    type CallFuture = BoxFuture<Option<Output>>;
+
+    fn on_call<Next, NextFuture>(
+        &self,
+        call: Call,
+        meta: M,
+        next: Next,
+    ) -> Either<Self::CallFuture, NextFuture>
+    where
+        Next: Fn(Call, M) -> NextFuture + Send + Sync,
+        NextFuture: Future<Output = Option<Output>> + Send + 'static,
+    {
+        let call_description = self.call_description(&call);
```

Comment:
Performance: Some calls contain a lot of data. Can we avoid calling `call_description` unless there is actually an error?

Reply:
That's not possible because `call` is moved into the call to `next`.

Reply:
`Call` implements `Clone`, so we could clone it before passing it to `next`.

(The Clone is likely to be cheaper and take less memory than the serialized `call_description`.)

Happy to do whatever you want here, and fix any performance issues if they happen.

Reply:
Done. I'm not sure if this will improve things, because I can't really tell if it's best to clone the type (which has a few heap allocated sub-types, like strings and possibly maps) or to build a string based on those same types :/

But as you said, this can be tweaked later if we hit any performance issues.